### PR TITLE
feat: add cash payment support with field mapping

### PR DESCRIPTION
**Pull request by Fenho**
## Summary

- Add support for cash payments in the payment intents API
- Implement field mapping from `payment_method` (SDK interface) to `payment_type` (API field)
- Add comprehensive TypeScript types for payment methods and requests
- Fix typo in paymentIntent.ts (`sernder_account` → `sender_account`)
- Add extensive test coverage for all payment field scenarios
- Update README with cash payment example

### 1. Enhanced PaymentIntentsManager
- Added field transformation logic that maps `payment_method` → `payment_type` before API calls
- Maintains backward compatibility while ensuring API consistency
- Supports both field names with `payment_method` taking precedence when both are provided

### 2. TypeScript Type Definitions
- Added `PaymentMethod` enum with `CASH` and `BANK_TRANSFER` values
- Created `PaymentIntentRequest` interface supporting both `payment_method` and `payment_type` fields
- Added `PaymentIntentResponse` interface matching API response structure

### 3. Comprehensive Test Coverage
- `payment_method: 'cash'` → API receives `payment_type: 'cash'`
- `payment_method: 'bank_transfer'` → API receives `payment_type: 'bank_transfer'`
- `payment_type: 'cash'` → API receives `payment_type: 'cash'` (unchanged)
- No payment method specified → API receives no payment type field
- Both fields provided → `payment_method` takes precedence

### 4. Documentation Updates
- Added cash payment example to README
- Updated create method documentation with cash payment usage

## Usage Example

```javascript
// Preferred SDK interface using payment_method
const paymentIntent = await fintocClient.paymentIntents.create({
  amount: 1000,
  currency: 'MXN',
  payment_method: 'cash'
});

// Backward compatibility - direct payment_type usage
const paymentIntent = await fintocClient.paymentIntents.create({
  amount: 1000,
  currency: 'MXN',
  payment_type: 'cash'
});
```

## Test Plan

- [x] All existing tests pass (171 tests)
- [x] New tests for cash payment creation
- [x] Tests for field mapping behavior
- [x] Tests for edge cases (both fields provided, no fields provided)
- [x] Tests for backward compatibility with direct `payment_type` usage
- [x] TypeScript compilation successful
- [x] Integration tests verify API request structure

🤖 Generated with [Claude Code](https://claude.ai/code)

### CODE REVIEW

File: src/lib/managers/paymentIntentsManager.ts [hunk 1]
```diff
@@ -1,7 +1,26 @@
+import { ResourceArguments } from '../../types';
 import { ManagerMixin } from '../mixins';
 import { PaymentIntent } from '../resources/paymentIntent';
 
 export class PaymentIntentsManager extends ManagerMixin<PaymentIntent> {
   static resource = 'payment_intent';
   static methods = ['list', 'get', 'create'];
+
+  protected async _create(args?: ResourceArguments): Promise<PaymentIntent> {
```
Comment (lezorich) [inline src/lib/managers/paymentIntentsManager.ts:9]: No entiendo porqué esto es necesario

File: src/types/paymentTypes.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/* eslint-disable camelcase, no-shadow */
+export enum PaymentMethodType {
+  CASH = 'cash',
+  BANK_TRANSFER = 'bank_transfer',
+}
+
+export interface PaymentIntentRequest {
```
Comment (lezorich) [inline src/types/paymentTypes.ts:7]: Esto porqué es necesario?

File: src/types/paymentTypes.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  amount: number;
+  currency: string;
+  payment_method?: PaymentMethodType;
+  payment_type?: PaymentMethodType;
+}
+
+export interface PaymentAccount {
```
Comment (lezorich) [inline src/types/paymentTypes.ts:14]: Y esto?

File: src/types/paymentTypes.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  type: string;
+  number: string;
+  holder_id: string;
+  institution_id: string;
+}
+
+export interface PaymentIntentResponse {
```
Comment (lezorich) [inline src/types/paymentTypes.ts:21]: Y esto?

**Review by lezorich — CHANGES_REQUESTED** [review]

File: src/lib/managers/paymentIntentsManager.ts (hunk 1 above)
Comment (Fenho) [reply src/lib/managers/paymentIntentsManager.ts:9]: porque necesitamos que tanto payment_method como payment_type funcionen para la llamada. En los docs hablamos de payment_type, y es lo que espera la API.
Como hay otros recursos que usan payment_method me parece bien que la DX no sea afectada y se acepten ambos

File: src/spec/integration.spec.ts [hunk 5]
```diff
@@ -62,15 +62,31 @@ test('fintoc.paymentIntents.create()', async (t) => {
...
 
   t.is(paymentIntent.method, 'post');
   t.is(paymentIntent.url, 'v1/payment_intents');
   t.is(paymentIntent.json.amount, paymentData.amount);
   t.is(paymentIntent.json.currency, paymentData.currency);
-  t.is(paymentIntent.json.payment_method, paymentData.payment_method);
+  t.is(paymentIntent.json.payment_type, paymentData.payment_type);
+});
+
+test('fintoc.paymentIntents.create() - cash payment', async (t) => {
```
Comment (lezorich) [inline src/spec/integration.spec.ts:76]: Esto no es necesario! No testea nada que no está testado en otro lado

**Review by lezorich — APPROVED** [review]

File: src/spec/integration.spec.ts (hunk 5 above)
Comment (Fenho) [reply src/spec/integration.spec.ts:76]: verdad
